mbedtls/src/pk/mod.rs

```diff
@@ -498,7 +506,21 @@ impl Pk {
         plain: &mut [u8],
         rng: &mut F,
     ) -> Result<usize> {
+        let ctx = self.inner.pk_ctx as *mut rsa_context;
```

Comment:
You need to check this is an RSA key first

Reply:
This `self.pk_type() == Type::Rsa` check would be sufficient. Right?